### PR TITLE
Azure AD Workload Identity authentication support

### PR DESCRIPTION
**Pull request by kostrse**
This PR adds support for Azure AD Workload Identity which is used for access to AAD service identities in Kubernetes.

This allows users who host Grafana in Kubernetes be able to use Workload Identities as alternative to Managed Identities which are not directly accessible by pods in Kubernetes.

Details on Azure AD Workload Identity:
* https://azure.github.io/azure-workload-identity
* https://github.com/Azure/azure-workload-identity

The newly introduced credentials type is `AzureWorkloadIdentityCredentials`.

Intended to be serialized in datasource JSON as:
```json
{
    "azureCredentials" {
        "authType": "workloadidentity"
    }
}
```

The credentials object doesn't provide any configuration options on the datasource side, but on the level it will be possible to override Tenant ID, Client ID, and Token File path.

Proposed configuration settings in Grafana config (not part of this PR):
```ini
[azure]
workload_identity_enabled = true

# These settings override default Workload Identity and normally not required
workload_identity_tenant_id =
workload_identity_client_id =
workload_identity_token_file =
```

### CODE REVIEW

File: azcredentials/credentials.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package azcredentials
 const (
 	AzureAuthCurrentUserIdentity = "currentuser"
 	AzureAuthManagedIdentity     = "msi"
+	AzureAuthWorkloadIdentity    = "wi"
```
Comment (aangelisc) [inline azcredentials/credentials.go:6]: Lets be more verbose here and change this to `workloadidentity` so it's very clear

File: aztokenprovider/retriever_wi.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+type workloadIdentityTokenRetriever struct {
+	tenantId   string
+	clientId   string
+	credential azcore.TokenCredential
+}
+
+func getWorkloadIdentityTokenRetriever(settings *azsettings.AzureSettings, credentials *azcredentials.AzureWorkloadIdentityCredentials) TokenRetriever {
+	var tenantId, clientId string
+	// TODO: See https://azure.github.io/azure-workload-identity/docs/faq.html#how-to-federate-multiple-identities-with-a-kubernetes-service-account
```
Comment (aangelisc) [inline aztokenprovider/retriever_wi.go:22]: Should we add support for this immediately?

Comment (kostrse) [reply aztokenprovider/retriever_wi.go]: Added

**Review by aangelisc — APPROVED** [review]
LGTM 🎉

**Pull request merged**